pkg/aws/cleaner.go

```diff
@@ -71,10 +73,22 @@ func deleteVpcAndEks(ctx context.Context, clients *clients, vpcID, clusterName s
 		return err
```

Comment:
I see we are using `multierr.Append` in some places, but we then return an error directly in some other cases.

I'd suggest making use of `multierr.Append` across the function and then exit early if desired.
```
logger.Info("listing VPC peering connections", "vpcID", vpcID)
    connections, err := listVpcPeeringConnections(ctx, clients.EC2, vpcID)
    if err != nil {
        errs = multierr.Append(errs, err)
       // return errs, if early exit is required
    } else {
        logger.Infof("deleting VPC peering connections: %d", len(connections))
        if err := deleteVpcPeeringConnections(ctx, clients.EC2, vpcID, connections); err != nil {
            errs = multierr.Append(errs, err)
        }
    }
```